### DISCUSSION TITLE
Some questions about MMDet 3.x visualizer

### DISCUSSION THREAD

**Discussion by sanbuphy** [open-mmlab/mmdetection#9173] [Q&A] [answered]
I am trying make sense the pipeline of  constructing a visualizer object. Here are my questions and thoughts.

1. What is the reason why the 'dataset_meta' needs to be used in the visualizer instance construction? Is this necessary?
`visualizer = VISUALIZERS.build(model.cfg.visualizer)`
`visualizer.dataset_meta = model.dataset_meta`
2. `add_datasample` is the method of visualizer(made in `mmdet/visualization/local_visualizer.py`) .A visualization of an detected image itself does not need initialize `dataset_meta`,It can show the detected image and draw box info. My guess is that if you need to draw gt_box, then you need to get labels by `dataset_meta`, so we need initialize the `datase_meta` in the begin.
3. when i run the visualizer it will shows:"mmengine/visualization/visualizer.py:170: UserWarning: `Visualizer` backend is not initialized because save_dir is None. warnings.warn('`Visualizer` backend is not initialized '       
 I think it means the visualizer need a save_dir in metaclass (MMEngine), but in `mmdet/visualization/local_visualizer.py`——DetLocalVisualizer class we had use `save_dir: Optional[str] = None,` to rewrite this method, is it necessary hint or omissions？

**Comment by ZwwWayne** [comment, marked as answer]
For question 1: Yes, this is necessary. The visualizer needs to obtain the class names and their color plate to show the category of boxes with consistent colors. All these information is about the dataset.
For question 2: for predictions, we also need to set the dataset meta to show the category names rather than labels.
For question 3: these might be a legacy issue to be fixed in mmdet, please @hhaAndroid to take a look

**Reply by sanbuphy** [reply]
Thank you for your reply！
For question 2,  I found that it can be displayed without setting dataset meta.
For question 3, I will ask him for help, Thanks!
